File: 13_14/Java/Assignment 6/Loan.java
package banking;
public abstract class Loan {

	private double principle;
	private int period;

	public Loan() {
	}

	public Loan(double p, int n) {
		principle = p;
		period = n;
	}

	public double getPrinciple() {
		return principle;
	}

	public void setPrinciple(double p) {
		principle = p;
	}

	public int getPeriod() {
		return period;
	}

	public void setPeriod(int n) {
		period = n;
	}

	 public abstract double getRate();


	public double getEmi() {
		double emi = 0;
		return (principle * ( 1 + (getRate() * period / 100)/ (12 * period)));
	}
}

		

File: Assignment5/LoanLib/Loan.cs
namespace LoanLib;

 public abstract class Loan
{
   private double principle;
   private int period;

   public Loan(double p, int n)
   {
      principle = p;
      period = n;
   }

    public double Principle
    {
        get {return principle;}

        set { principle = value; }
   }
      
    public int Period
    {
        get {return period;}
        set { period = value;}
    }
    
    public abstract double Rate();

    public double Emi()
    { 
        double emi = 0;
     return emi = principle * (1 + Rate() * period / 100) / (12 * period);
    }
    
}
